### PR TITLE
SDL_compat: Fix on Darwin

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes

- Pull in upstream patch for correct dylib install name.
- Port rpath patching to Darwin

Tested by building & running DOSBox.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
Result of `nixpkgs-review pr 190390` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>SDL_compat</li>
    <li>dosbox</li>
    <li>keen4</li>
    <li>open-watcom-v2</li>
    <li>open-watcom-v2-unwrapped</li>
    <li>openxcom</li>
    <li>rott</li>
    <li>rott-shareware</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 190390` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>SDL_compat</li>
    <li>dosbox</li>
    <li>keen4</li>
    <li>open-watcom-v2</li>
    <li>open-watcom-v2-unwrapped</li>
    <li>rott</li>
    <li>rott-shareware</li>
  </ul>
</details>

---

`dosbox` launches on both. `open-watcom-v2` builds on Darwin but is broken & tests fail, but that's unrelated to SDL_compat: just an upstream change I noticed and can patch now that `dosbox` works again.

File: pkgs/development/libraries/SDL_compat/default.nix [hunk 1]
```diff
@@ -25,25 +28,39 @@ stdenv.mkDerivation rec {
...
+    ++ optionals stdenv.hostPlatform.isDarwin [ libiconv Cocoa ]
     ++ optionals openglSupport [ libGL libGLU ];
 
   enableParallelBuilding = true;
 
   setupHook = ../SDL/setup-hook.sh;
 
   postFixup = ''
-    for lib in $out/lib/*.so* ; do
+    for lib in $out/lib/*${stdenv.hostPlatform.extensions.sharedLibrary}* ; do
```
Comment (peterhoeg) [inline pkgs/development/libraries/SDL_compat/default.nix:52]: Shouldn't this be `targetPlatform` instead/

Comment (OPNA2608) [reply pkgs/development/libraries/SDL_compat/default.nix:52]: I haven't specifically tested cross-compiling, where the nuances of `*Platform` would become relevant, but `targetPlatform` should only be relevant when compiling something that produces something for another platform (like a compiler).

https://nixos.org/manual/nixpkgs/stable/#ssec-cross-platform-parameters

`buildPlatform` is the platform we're building X on, `hostPlatform` the platform we're building X for and `targetPlatform` the platform X itself builds for. To my understanding, `host` should be correct one here.

Comment (peterhoeg) [reply pkgs/development/libraries/SDL_compat/default.nix:52]: I stand corrected!

**Pull request merged**